Doc/library/typing.rst

```diff
@@ -648,6 +648,9 @@ to this is that a list of types can be used to substitute a :class:`ParamSpec`::
    >>> Z[int, [dict, float]]
    __main__.Z[int, [dict, float]]
 
+   >>> 1 + 2
+   4
```

Comment:
this isn't actually doctested, I don't think, since you don't have the `.. doctest::` directive above this block

Reply:
It is: 

```
» ./python.exe -m doctest Doc/library/typing.rst
**********************************************************************
File "Doc/library/typing.rst", line 651, in typing.rst
Failed example:
    1 + 2
Expected:
    4
Got:
    3
**********************************************************************
1 items had failures:
   1 of  48 in typing.rst
***Test Failed*** 1 failures.
```

It works the same way with and without `.. doctest::`.
The problem might that Sphinx does not recognize doctests without this directive, but this is clearly a Sphinx's own issue.

Reply:
Yes, that was what I meant, sorry for not being clear: that it wouldn't be picked up by sphinx's doctest job :)

Reply:
We might need to use `doctest_test_doctest_blocks` from https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

Reply:
> We might need to use `doctest_test_doctest_blocks` from https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

We can try enabling that, sure, but that should be a separate issue/PR. This PR is about fixing broken doctests in `typing.rst`, not changing the way we determine in CI which doctests should be run. The majority of these that need to be tested already are tested, using the `.. doctest::` directive.